### ISSUE TITLE
Mybinder failed to connect to ipykernel on Safari

### ISSUE THREAD

**Issue by bitinn** [jupyterhub/mybinder.org-deploy#2278]
Using Safari 15.4 on macOS 12.3, this is what I see (the reconnection repeat every second but never connects):

<img width="443" alt="Screen Shot 2022-06-13 at 20 46 37" src="https://user-images.githubusercontent.com/1484279/173357474-c4712465-d0c2-40c7-be53-693dcfff473b.png">

Using Chrome on the same Mac, using the same network, I am able to connect correctly:

<img width="559" alt="Screen Shot 2022-06-13 at 20 46 45" src="https://user-images.githubusercontent.com/1484279/173357642-8fa17c59-e728-43e3-86be-ff4aa206e802.png">

Maybe worth investigating why? I never managed to use mybinder via Safari.

**Comment by bitinn** [comment]
Hi I was using this repo, you can see if this can be reproduced on your end:

https://github.com/gboeing/osmnx-examples

happy to try out other repo if you need.

**Comment by MridulS** [comment]
I just tried running the couple of notebooks from the repo on mybinder.org and it seems to work just fine.
I am using Safari 15.5 if that helps :)

Could you try running https://mybinder.org/v2/gh/binder-examples/requirements/HEAD, does this work on your machine?

Also are you behind a corporate VPN? This could very well be an issue on the jupyterlab side too and I doubt this is a mybinder.org issue.

**Comment by bitinn** [comment]
> I just tried running the couple of notebooks from the repo on mybinder.org and it seems to work just fine. I am using Safari 15.5 if that helps :)
> 
> Could you try running https://mybinder.org/v2/gh/binder-examples/requirements/HEAD, does this work on your machine?
> 
> Also are you behind a corporate VPN? This could very well be an issue on the jupyterlab side too and I doubt this is a mybinder.org issue.

Tried the example repo, no luck, same issue.

I tried both with or without VPN, same issue. Network issue doesn't explain the Chrome situation however.

**Comment by bitinn** [comment]
Let me know if you can spot any potential issue fixed in 15.5:

https://developer.apple.com/documentation/safari-release-notes/safari-15_5-release-notes

**Comment by manics** [comment]
@bitinn Can you open your browser developer/web console and paste all logs? e.g. see https://support.monday.com/hc/en-us/articles/360002197259-How-to-Open-the-Developer-Console

**Comment by bitinn** [comment]
<img width="1381" alt="Screen Shot 2022-06-14 at 20 50 44" src="https://user-images.githubusercontent.com/1484279/173581699-b78b3ec9-3dca-44ba-a12c-d79515ce9936.png">

@manics  Just endless stream of these errors

**Comment by MridulS** [comment]
Yeah that's a network connection problem or there is a safari extensions that's messing with the requests. Could you disable all extensions on Safari?

**Comment by manics** [comment]
In addition could you tell us whether you have:
- firewalls running locally or on your network
- any network proxies (including transparent proxies)
- anti-virus running locally

It's possible they're manipulating the connection in a way that Chrome can handle but Safari doesn't

**Comment by bitinn** [comment]
An update: today I was in office using the same network but on an older macOS (10.15.7), with Safari 15.3;

<img width="572" alt="Screen Shot 2022-06-24 at 15 15 11" src="https://user-images.githubusercontent.com/1484279/175483476-8b684959-ca92-4838-8b95-ebe18906cc88.png">

The connection works, so I highly highly suspect this has something to do with macOS / Safari updates (to be honest this is not the first time I see a site that worked perfectly fine breaks in latest Safari).
